src/modelize/modelize_property.nit

```diff
@@ -218,6 +222,50 @@ redef class ModelBuilder
 			return
 		end
 
+		# Look at the autoinit class-annotation
+		var autoinit = nclassdef.get_single_annotation("autoinit", self)
+		if autoinit != null then
+			# Just throws the collected initializers
+			mparameters.clear
+			initializers.clear
+
+			# Get and check each argument
+			for narg in autoinit.n_args do
+				var id = narg.as_id
+				if id == null then
+					error(narg, "Syntax error: \"autoinit\" expects method identifiers.")
+					return
+				end
+
+				# Search the property.
+				# To avoid bad surprises, try to get the setter first.
+				var p = try_get_mproperty_by_name(narg, mclassdef, id + "=")
+				if p == null then
+					p = try_get_mproperty_by_name(narg, mclassdef, id)
+				end
+				if p == null then
+					error(narg, "Error: unknown method `{id}`")
+					return
+				end
+				if not p.is_autoinit then
+					error(narg, "Error: `{p}` is not an autoinit method")
+					return
+				end
+
+				# Register the initializer and the parameters
+				initializers.add(p)
+				var pd = p.intro
+				if pd isa MMethodDef then
+					# Get the signature resolved for the current receiver
+					var sig = pd.msignature.resolve_for(mclassdef.mclass.mclass_type, mclassdef.bound_mtype, mclassdef.mmodule, false)
+					mparameters.add_all sig.mparameters
+				else
+					abort
```

Comment:
This should be simplified, right?

~~~nit
if not pd isa MMethodDef then abort
~~~